lib/util/utils.js

```diff
@@ -759,42 +759,58 @@ function isPropertyRequired(propName, model) {
  */
 exports.statusCodeStringToStatusCode = lodash.invert(lodash.mapValues(http.STATUS_CODES, value => { return value.replace(/ |-/g, "").toLowerCase(); }));
 
-
 /**
- * Models an ARM cloud error.
+ * Models an ARM cloud error schema.
  */
-exports.CloudError = {
+exports.CloudErrorSchema = {
   description: "Error response describing why the operation failed.",
   schema: {
-    properties: {
-      error: {
-        properties: {
-          code: {
-            type: "string",
-            description: "An identifier for the error. Codes are invariant and are intended to be consumed programmatically."
-          },
-          message: {
-            type: "string",
-            description: "A message describing the error, intended to be suitable for display in a user interface."
-          },
-          target: {
-            type: "string",
-            description: "The target of the particular error. For example, the name of the property in error."
-          },
-          details: {
-            type: "array",
-            items: { type: "object" },
-            description: "A list of additional details about the error."
-          },
-          additionalInfo: {
-            type: "array",
-            items: { type: "object" },
-            description: "A list of additional info about an error."
-          }
-        },
-        additionalProperties: false
-      }
-    },
-    additionalProperties: false
+    "$ref": "#/definitions/CloudErrorWrapper"
   }
+};
+
+/**
+ * Models an ARM cloud error wrapper.
+ */
+exports.CloudErrorWrapper = {
+  type: "object",
+  properties: {
+    error: {
+      "$ref": "#/definitions/CloudError"
+    }
+  },
+  additionalProperties: false
+};
+
+/**
+ * Models a Cloud Error
+ */
+exports.CloudError = {
+  type: "object",
+  properties: {
+    code: {
+      type: "string",
+      description: "An identifier for the error. Codes are invariant and are intended to be consumed programmatically."
+    },
+    message: {
+      type: "string",
+      description: "A message describing the error, intended to be suitable for display in a user interface."
+    },
+    target: {
+      type: "string",
+      description: "The target of the particular error. For example, the name of the property in error."
+    },
+    details: {
+      type: "array",
+      items: { type: "object" },
+      description: "A list of additional details about the error."
+    },
+    additionalInfo: {
```

Comment:
wanna add innerError as well?

Reply:
yes just that the spec is a little confusing. It says the type is string but it seems to be an object. I'll just leave it without specifying any type.

Reply:
ah nevermind, it seems to be always an object

Reply:
done